pandas/tests/frame/indexing/test_mask.py

```diff
@@ -150,3 +151,12 @@ def test_mask_inplace_no_other():
     df.mask(cond, inplace=True)
     expected = DataFrame({"a": [np.nan, 2], "b": ["x", np.nan]})
     tm.assert_frame_equal(df, expected)
+
+
+def test_mask_with_na():
+    # See GH #52955, NA should propagate in mask
+    df = DataFrame([[1, NA], [NA, 2]], dtype=Int64Dtype())
+    result = df.mask(df % 2 == 1, 0)
+
+    expected = DataFrame([[0, NA], [NA, 2]], dtype=Int64Dtype())
+    tm.assert_frame_equal(result, expected)
```

Comment:
Can you also do a test for:

```python
result = df.mask(df[0] % 2 == 1, 0)
```

or similar I.e. test for a series as the mask also. Same below for `where` .

Reply:
Sure!